API_DOCS.md

```diff
@@ -125,6 +130,12 @@ Describes the shape and behaviour of the resources object you will pass to `getL
 | `commaSeparatedBatchKey` | (Optional) Set to true if the interface of the resource takes the batch key as a comma separated list (rather than an array of IDs, as is more common). Default: false                                                                                   |
 | `isResponseDictionary`   | (Optional) Set to true if the batch resource returns the results as a dictionary with key mapped to values (instead of a list of items). If this option is supplied `reorderResultsByKey` should not be. Default: false                                  |
 | `isBatchKeyASet`         | (Optional) Set to true if the interface of the resource takes the batch key as a set (rather than an array). For example, when using a generated clientlib based on swagger where `uniqueItems: true` is set for the batchKey parameter. Default: false. |
+| `propertyBatchKey`       | (Optional) The argument to the resource that represents the optional properties we want to fetch. (e.g. usually 'properties' or 'features').                                                                                                             |
+| `responseKey`            | (Non-optional when propertyBatchKey is used) The key in the response objects corresponds to `batchKey`. This should be the only field that are marked as required in your swagger endpoint response, except nestedPath.                                  |
```

Comment:
is this the same as `newKey`? looks like they share a similar purpose and will likely be the same values. Can we reuse (maybe rename) this?

Reply:
They might be the same, or completely different! 
`newKey` is the singular version of the `batchKey`.
However, `responseKey` is the key correspond to `batchKey` in the swagger response, we cannot control what it is. 

For example, we have this swagger endpoint, input ags
```
user_ids: str[] 
properties: str[] 
```
response: 
```
{
id: str,
name: str,
age: number,
....
}
```
batchKey is `user_ids` and newKey is `user_id` but responseKey is `id`, which is different from newKey.